cmds/core/cpio/cpio_test.go

```diff
@@ -79,18 +81,25 @@ func TestCpio(t *testing.T) {
 		}
 	}
 
-	c := testutil.Command(t, "-v", "o")
-	c.Dir = tempDir
+	inputFile, err := os.CreateTemp(tempDir, "")
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+	defer os.Remove(inputFile.Name())
+	defer inputFile.Close()
 
-	buffer := bytes.Buffer{}
 	for _, ent := range targets {
-		buffer.WriteString(ent.Name + "\n")
+		name := filepath.Join(tempDir, ent.Name)
+		if _, err := fmt.Fprintln(inputFile, name); err != nil {
+			t.Fatalf("failed to write file path %v to input file: %v", ent.Name, err)
+		}
 	}
-	c.Stdin = &buffer
+	inputFile.Seek(0, 0)
 
-	archive, err := c.Output()
+	archive := &bytes.Buffer{}
+	err = run([]string{"o"}, inputFile, archive, true, "newc")
 	if err != nil {
-		t.Fatalf("%s %v", c.Stderr, err)
+		t.Fatalf("failed to build archive from filepaths: %v", err)
 	}
 
 	// Cpio can't read from a non-seekable input (e.g. a pipe) in input mode.
```

Comment:
I just now noticed this comment from an old commit. It makes no sense to me: cpio should alway be able to read from a pipe. Do you understand it? I don't I'm afraid. Thanks.

Reply:
Good catch, you are right. I checked out the [commit](https://github.com/RiSKeD/u-root/commit/301d69f0ab67540205bed49b191433a4c8456fe8) for this comment and reading from pipes was only latter introduced in this [commit](https://github.com/RiSKeD/u-root/commit/665d97ec19db45550c001223c69e2acb7e41860c). 
So this restriction is not longer true. I think it would be best to create a separate test reading from a pipe as well.

Reply:
It is pretty easy to use an io.Pipe, which is just an in-memory implementation of a pipe, and I think the error usage of the test would have no trouble with it. If you could take a look at doing that it would be very helpful.

Reply:
let me know your plan -- if you'd like this to go in as is.

Reply:
Thanks for your review. I rebased the PR afterward to integrate the small changes for the error wrapping. I hope I did everything right as intended.

Reply:
hey, one last thing: can you follow abrender's idea and use t.TempDir? I realize this is "just one more thing" but it does simplify the test code?

Reply:
Sure, I will add his suggestion.